src/main/webapp/courseLoader.js

```diff
@@ -81,97 +92,103 @@
+    };
 
-    const currentTermRatingAvg = average(
-      /* adds dummy data */ [4, 8, 17].concat(termDataObject.termScoreList)
-    );
-    const prevTermRatingAvg = average(
-      /* adds dummy data */ [12, 5, 3].concat(prevTermData1.termScoreList)
+    const chart = new google.charts.Bar(
+      document.getElementById("term-rating-comp")
     );
-    const prevTermRatingAvg2 = average(
-      /* adds dummy data */ [10, 8, 6].concat(prevTermData2.termScoreList)
+    chart.draw(comparisonData, google.charts.Bar.convertOptions(options));
+  }
+
+  async function makeTermPerceptionChart(termDataObject) {
+    const average = (list) =>
+      list.reduce((prev, curr) => prev + curr) / list.length;
+    const currentPerceptionRatingAvg = average(
+      /* adds dummy data */ [21, 11, 9].concat(
+        termDataObject.termPerceptionList
+      )
     );
 
-    const comparisonData = google.visualization.arrayToDataTable([
-      ["year", `${currentTerm[0]}  ${currentTerm[1]}`, prevTerm1, prevTerm2],
-      [
-        currentTerm[1],
-        currentTermRatingAvg,
-        prevTermRatingAvg,
-        prevTermRatingAvg2,
-      ],
+    const prevTermData = [];
+    const avgData = [];
+    const prevTermNameList = await getPrevTermName(2);
+
+    for (let term of prevTermNameList) {
+      prevTermData.push(await getPreviousTermData(term));
+    }
+
+    for (let termData of prevTermData) {
+      avgData.push(average(termData.termPerceptionList.concat([17, 8, 5])));
+    }
+
+    const perceptionData = google.visualization.arrayToDataTable([
+      [" ", term].concat(prevTermNameList),
+      [" ", currentPerceptionRatingAvg].concat(avgData),
     ]);
 
     const options = {
-      title: "Average Term Rating Comparison",
+      title: "Average Term Perception Comparison",
       height: 450,
       bars: "horizontal",
       bar: { groupWidth: "30%" },
-      hAxis: { title: "Avg Term Rating" },
+      hAxis: { title: "Avg Term Perception" },
     };
 
     const chart = new google.charts.Bar(
-      document.getElementById("term-rating-comp")
+      document.getElementById("term-perception-comp")
     );
-    chart.draw(comparisonData, google.charts.Bar.convertOptions(options));
+    chart.draw(perceptionData, google.charts.Bar.convertOptions(options));
   }
 
   async function getPreviousTermData(prevTerm) {
     const url = `/term-live?school-name=${schoolName}&course-name=${courseName}&term=${prevTerm}&prof-name=${profName}&num-units=${units}`;
     const response = await fetch(url);
+    return response.json();
+  }
+
+  async function getPrevTermName(comparisonCount) {
+    const url = `/prev-terms?school-name=${schoolName}&course-name=${courseName}&term=${term}&prof-name=${profName}&num-units=${units}&comparison-count=${comparisonCount}`;
+    const response = await fetch(url);
     const prevTermData = await response.json();
-    return prevTermData;
+    const [prevTermName1, prevTermName2] = [
```

Comment:
This is still hard coding that you'll only ever deal with 2 returned terms. You can make this more flexible by doing something like:
```
return prevTermData.map((data) => data.properties.term);
```

It's shorter this way too : )

Reply:
oh this is a bug forgot to change this to just a list

Reply:
Nice catch I forgot to change the return!